### PR TITLE
[Merged by Bors] - feat(KrullDimension): height_eq_coe_iff_minimal_le_height

### PR DESCRIPTION
**Pull request by nomeata**
The elements of finite height `n` are the minimial elements among those of height `≥ n`.

From the Carleson project

### CODE REVIEW

File: Mathlib/Order/KrullDimension.lean [hunk 1]
```diff
@@ -318,6 +319,21 @@ lemma height_eq_coe_iff {x : α} {n : ℕ} :
     rename_i y _
     cases height y <;> simp; norm_cast; omega
 
+/-- The elements of finite height `n` are the minimial elements among those of height `≥ n`. -/
+lemma height_eq_coe_iff_minimal_le_height (a : α) (n : ℕ) :
```
Comment (YaelDillies) [inline Mathlib/Order/KrullDimension.lean:323]:
```suggestion
lemma height_eq_coe_iff_minimal_le_height {a : α} {n : ℕ} :
```

File: Mathlib/Order/KrullDimension.lean [hunk 2]
```diff
@@ -318,6 +319,21 @@ lemma height_eq_coe_iff {x : α} {n : ℕ} :
...
+    height a = n ↔ Minimal (fun y => n ≤ height y) a := by
+  by_cases hfin : height a < ⊤
+  · cases hn : n with
+    | zero => simp
+    | succ => simp [minimal_iff_forall_lt, height_eq_coe_add_one_iff, ENat.add_one_le_iff,
+        coe_lt_height_iff, *]
+  · suffices ∃ x, ∃ (_ : x < a), ↑n ≤ height x by
```
Comment (YaelDillies) [inline Mathlib/Order/KrullDimension.lean:330]:
```suggestion
  · suffices ∃ x < a, ↑n ≤ height x by
```
works, I would think

File: Mathlib/Order/KrullDimension.lean (hunk 1 above)
Comment (nomeata) [reply Mathlib/Order/KrullDimension.lean]: Ha, seems my grepping for `_iff (` was insufficient.

**Review by YaelDillies — APPROVED** [review]
WIll you add the coheight lemma later?

maintainer merge

**Comment by nomeata** [comment]
Yes, once I got all the lemmas through the review I'll do all the corresponding coheight lemmas in one go.

**Review by TwoFX — APPROVED** [review]
Thanks!
bors r+

**Pull request closed without merging**